runtime/pkg/conncache/conncache.go

```diff
@@ -136,6 +138,25 @@ func New(opts Options) Cache {
 		go c.periodicallyCheckHangingConnections()
 	}
 
+	go func() {
+		for {
+			time.Sleep(5 * time.Second)
+			go func() {
```

Comment:
This is missing cancellation from `Close`. Instead of having a background loop (which adds concurrency complexity), how about just checking the TTL when accessing a cached error in `Acquire`? That should still ensure correctness, and cleanup for connections that are not acquired again anyway happens automatically by the LRU cache

Reply:
Done.